mkt/developers/views.py

```diff
@@ -373,12 +374,32 @@ def content_ratings_edit(request, addon_id, addon):
     }
     request.session.modified = True
 
-    return render(request, 'developers/apps/ratings/ratings_edit.html',
-                  {'addon': addon,
-                   'app_name': get_iarc_app_title(addon),
-                   'form': form,
-                   'company': addon.latest_version.developer_name,
-                   'now': datetime.now().strftime('%Y-%m-%d %H:%M:%S')})
+    ctx = {
+        'addon': addon,
+        'app_name': get_iarc_app_title(addon),
+        'form': form,
+        'company': addon.latest_version.developer_name,
+        'now': datetime.now().strftime('%Y-%m-%d %H:%M:%S')
+    }
+
+    if waffle.switch_is_active('iarc-upgrade'):
+        iarc_request_id = IARCRequestID.objects.filter(app_id=addon)
+        if iarc_request_id.exists():
+            iarc_request_id = iarc_request_id[0]
+            outdated = (datetime.now() - iarc_request_id.created >
+                        timedelta(hours=1))
+            if outdated:
+                # IARC Request ID outdated. Re-create.
```

Comment:
The flow is asynchronous and the request ID is just a way of us not having to pass in all the app info to IARC. We give them an ID, and then in the content ratings pingback, we use that ID to reference back to the app.